src/port.js

```diff
@@ -114,6 +141,23 @@ export default class Port {
 		}
 
 	}
+	registerService(serviceType, version, service) {
+		if(this.isConnected) {
+			throw new Error('Register services before connecting');
+		}
+		if(!typeNameValidator.test(serviceType)) {
+			throw new Error(`Invalid service type "${serviceType}"`);
+		}
+		let methodNames = [];
+		for(let p in service) {
+			if(typeof(service[p]) === 'function') {
+				methodNames.push(p);
+				this.onRequest(`service:${serviceType}:${version}:${p}`, service[p]);
```

Comment:
Confirmed that this doesn't actually explode. Registering the service goes through all the methods and adds a request handler for each one, passing in a reference to each function. Setting the original service property to null after this happens has no effect.